src/main/java/seedu/todo/controllers/ClearController.java

```diff
@@ -160,131 +193,102 @@ else if (dateOn != null) {
-            }
-        }
-        if (parsedDate != null) {
-            if (parsedDate[0] != null) {
-                commandLineMessage = String.format("%s by <date>", commandLineMessage);
-            } else if (parsedDate[1] != null && parsedDate[2] != null) {
-                commandLineMessage = String.format("%s from <date> to <date>", commandLineMessage);
-            } else if (parsedDate[1] != null) {
-                commandLineMessage = String.format("%s from <date>", commandLineMessage);
-            } else {
-                commandLineMessage = String.format("%s to <date>", commandLineMessage);
-            }
-        }
-        Renderer.renderDisambiguation(commandLineMessage, consoleDisplayMessage);
-    }
-    
-    /**
      * clear all tasks and events of the date that exist in the database.
      * 
      * @param TodoListDB
      * @param givenDate
      *            null if parsing failed or Due date for Task or start date for Event
+     * @param deleteAll
+     *            true if no CalendarItem Type provided, false if "task" or "event" keyword found
+     * @param isTask
+     *            true if "task" keyword found, false if "event" keyword found
      */
     private void destroyBySelectedDate(TodoListDB db, LocalDateTime givenDate, boolean deleteAll, boolean isTask) {
         int numTasks = db.getTaskByDate(givenDate).size();
         int numEvents = db.getEventByDate(givenDate).size();
-        if (numTasks == 0 && numEvents == 0) {
+        
+        // no tasks or events are been found
+        if (numTasks == EMPTY && numEvents == EMPTY) {
             Renderer.renderIndex(db, MESSAGE_CLEAR_NO_ITEM_FOUND);
             return;
         }
         
+        // task or event is not specified
         if (deleteAll) {
             db.destroyAllEventByDate(givenDate);
             db.destroyAllTaskByDate(givenDate);
-        } else if (isTask) {
-            if (numTasks == 0) {
+        } else if (isTask) { //deleting task
+            if (numTasks == EMPTY) { //if no task is found
                 Renderer.renderIndex(db, MESSAGE_CLEAR_NO_ITEM_FOUND);
                 return;
             }
             db.destroyAllTaskByDate(givenDate);
-            numEvents = 0;
-        } else {
-            if (numEvents == 0) {
+            numEvents = EMPTY;
+        } else { //deleting events
+            if (numEvents == EMPTY) { //if no event is found
                 Renderer.renderIndex(db, MESSAGE_CLEAR_NO_ITEM_FOUND);
                 return;
             }
             db.destroyAllEventByDate(givenDate);
-            numTasks = 0;
+            numTasks = EMPTY;
         }
+        
+        //save and render
         db.save();
-        Renderer.renderIndex(db, String.format(MESSAGE_CLEAR_SUCCESS, displaySuccessMessage(numTasks, numEvents)));
+        Renderer.renderIndex(db, String.format(MESSAGE_CLEAR_SUCCESS, formatSuccessMessage(numTasks, numEvents)));
```

Comment:
I think that nesting multiple `String.format`s together makes it unreadable and have to jump through many hoops to find what outputs here. `formatSuccessMessage` is also not a descriptive name for a method which returns "x tasks and y events".

My suggestion is that instead of declaring these methods privately in this controller, you either move them to StringUtil or you don't do so much abstraction.